crates/deltalake-core/src/operations/write.rs

```diff
@@ -960,4 +1132,177 @@ mod tests {
 
         assert_batches_eq!(&expected, &data);
     }
+
+    // TODO: add tests for replaceWhere with partition column
+
+    #[tokio::test]
+    async fn test_replace_where() {
+        // let tmp_dir = tempdir::TempDir::new("test").unwrap();
+        // let tmp_path = std::fs::canonicalize(tmp_dir.path()).unwrap();
+
+        let schema = get_arrow_schema(&None);
+        // let table_path = tmp_path.as_os_str().to_str().unwrap();
+        // let ops = DeltaOps::try_from_uri(table_path).await.unwrap();
+
+        let batch = RecordBatch::try_new(
+            Arc::clone(&schema),
+            vec![
+                Arc::new(arrow::array::StringArray::from(vec!["A", "B", "C", "C"])),
+                Arc::new(arrow::array::Int32Array::from(vec![0, 20, 10, 100])),
+                Arc::new(arrow::array::StringArray::from(vec![
+                    "2021-02-02",
+                    "2021-02-03",
+                    "2021-02-02",
+                    "2021-02-04",
+                ])),
+            ],
+        )
+        .unwrap();
+
+        let table = DeltaOps::new_in_memory()
+            .write(vec![batch])
+            .with_save_mode(SaveMode::Append)
+            .await
+            .unwrap();
+        assert_eq!(table.version(), 0);
+
+        let batch_add = RecordBatch::try_new(
+            Arc::clone(&schema),
+            vec![
+                Arc::new(arrow::array::StringArray::from(vec!["C"])),
+                Arc::new(arrow::array::Int32Array::from(vec![50])),
+                Arc::new(arrow::array::StringArray::from(vec!["2023-01-01"])),
+            ],
+        )
+        .unwrap();
+
+        let table = DeltaOps(table)
+            .write(vec![batch_add])
+            .with_save_mode(SaveMode::Overwrite)
+            .with_replace_where(col("id").eq(lit("C")))
+            .await
+            .unwrap();
+        assert_eq!(table.version(), 1);
+
+        let expected = [
+            "+----+-------+------------+",
+            "| id | value | modified   |",
+            "+----+-------+------------+",
+            "| A  | 0     | 2021-02-02 |",
+            "| B  | 20    | 2021-02-03 |",
+            "| C  | 50    | 2023-01-01 |",
+            "+----+-------+------------+",
+        ];
+        let actual = get_data(&table).await;
+        assert_batches_sorted_eq!(&expected, &actual);
+    }
+
+    #[tokio::test]
+    async fn test_replace_where_fail_not_matching_predicate() {
+        let tmp_dir = tempdir::TempDir::new("test").unwrap();
+        let tmp_path = std::fs::canonicalize(tmp_dir.path()).unwrap();
```

Comment:
Instead of writing to disk the suggested approach is to use a memory table.
See: https://github.com/delta-io/delta-rs/blob/main/crates/deltalake-core/src/operations/merge/mod.rs#L1527-L1528

Reply:
The reason I used a disk location was because I could not find a way to load the table from memory after a failed write op. Few lines liner I have `table.is_err()` so all my attempt to reload it after that were unsuccessful so I ended up using file system. Is there a way?

Reply:
It might be possible to create a clone of the table prior to failure then update the clone to the latest version after failure. not 100% sure on that.

Reply:
Cool, I'll give that a try

Reply:
@Blajda I tried few options as you suggested but to no avail. DeltaTable is not clone-able. The closest I could get was serializing/de-ser but looks like table memory is lost in the process so I can no longer refer to files (even though ` table.get_latest_version()` of the deserialized table returns a correct value, this is likely just a deserialised `self.version` value without touching the table in memory)